### DISCUSSION TITLE
Input field directives?

### DISCUSSION THREAD

**Discussion by jhay06** [mirumee/ariadne#1042] [Q&A]
Any good example how to use INPUT FIELD Directives ? i saw the documentation but the sample there is only for type field not for input field

Thanks

**Comment by rafalp** [comment]
Here's reference of all available visitor methods on directive: https://ariadnegraphql.org/docs/api-reference#example-schema-visitors

Your directive should define custom `visit_input_field_definition` method that will be called with input's field and input type:

```python
class MyDirective(SchemaDirectiveVisitor):
    def visit_input_field_definition(
        self, field: GraphQLInputField, object_type: GraphQLInputObjectType
    ) -> GraphQLInputField:
        # Do something with input's field
```
